src/main.js

```diff
@@ -5,20 +5,21 @@ import store from './store'
 
 Vue.config.productionTip = false
 
-fetch('/api/sessions')
-  .then(r => r.json())
-  .then(data => {
-    if (data && data.user) {
-      store.dispatch('login', data.user)
-    } else {
-      store.dispatch('loadApplicationSettings')
-    }
-    new Vue({
-      router,
-      store,
-      render: h => h(App)
-    }).$mount('#app')
-  })
-  .catch(e => {
-    console.log(e)
-  })
+async function main () {
+  const session = await fetch('/api/sessions')
+  const data = await session.json()
+
+  if (data && data.user) {
+    await store.dispatch('login', data.user)
+  } else {
+    await store.dispatch('loadApplicationSettings')
+  }
```

Comment:
this logic is still buggy. in particular, we only seem to set `viewPeriodId` in `loadApplicationSettings` and it ends up being `null` a lot of the time, which kinda still works because it should be set to the current period, and the backend generally also uses the current period if the specified period is not set, but this is not ideal. the entire flow of how the app loads it's initial state into the store should be re-worked.

Reply:
referenced in #234

Reply:
Yikes, thanks for flagging this.